chanutils/queue.go

```diff
@@ -0,0 +1,132 @@
+package chanutils
+
+import (
+	"container/list"
+	"sync"
+)
+
+const (
+	// DefaultQueueSize is the default size to use for concurrent queues.
+	DefaultQueueSize = 10
+)
+
+// ConcurrentQueue is a typed concurrent-safe FIFO queue with unbounded
+// capacity. Clients interact with the queue by pushing items into the in
+// channel and popping items from the out channel. There is a goroutine that
+// manages moving items from the in channel to the out channel in the correct
+// order that must be started by calling Start().
+type ConcurrentQueue[T any] struct {
```

Comment:
I'm assuming the package `module github.com/lightningnetwork/lnd/queue` is no longer maintained right?